cachito/web/api_v1.py

```diff
@@ -469,3 +479,32 @@ def add_request_config_files(request_id):
 
     db.session.commit()
     return "", 204
+
+
+@api_v1.route("/requests/<int:request_id>/logs")
+def get_request_logs(request_id):
+    """
+    Retrieve the logs for the Cachito request.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    :raise ValidationError: if the JSON is invalid
+    """
+    request_log_dir = flask.current_app.config["CACHITO_REQUEST_FILE_LOGS_DIR"]
+    if not request_log_dir:
+        raise NotFound()
+    request = Request.query.get_or_404(request_id)
+    log_file_path = os.path.join(request_log_dir, f"{request_id}.log")
+    if not os.path.exists(log_file_path):
+        if request.state.state_name == "stale":
+            raise Gone(f"The logs for the cachito request {request_id} no longer exist")
+        finalized = request.state.state_name in RequestStateMapping.get_final_states()
+        if finalized:
+            raise NotFound()
+        # The request may not have been initiated yet. Return empty logs until it's processed.
+        return flask.Response("", mimetype="text/plain")
+
+    with open(log_file_path) as f:
+        return flask.Response(f.read(), mimetype="text/plain")
```

Comment:
@lcarva when you implemented this in IIB, did you first try streaming the data to avoid Cachito loading it all in memory at once?

https://flask.palletsprojects.com/en/1.1.x/patterns/streaming/

It's likely not an issue now but debug logs tend to get quite noisy over time.

Reply:
No particular reason other than I didn't think of it. +1 to doing this here (and in IIB).